### PR TITLE
tests: reduce flake rates when apiserver is 503

### PR DESCRIPTION
**Pull request by mfojtik**
This introduce `withRetryOnTemporaryErr` for extended test client that retry when API server return temporary error that are not being retried by client-go (or increase the interval we retry).

### CODE REVIEW

File: test/extended/util/client.go [hunk 1]
```diff
@@ -702,18 +703,54 @@ func (c *CLI) AddResourceToDelete(resource schema.GroupVersionResource, metadata
...
+	}
+	if err := kretry.OnError(kretry.DefaultRetry, retriableAPIErrors, func() error {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("context cancelled")
+		default:
+		}
+		return fn()
+	}); err != nil {
+		FatalErr(err)
```
Comment (sttts) [inline test/extended/util/client.go:739]: how fatal  is this? Will be get proper error messages?

File: test/extended/util/client.go [hunk 2]
```diff
@@ -702,18 +703,54 @@ func (c *CLI) AddResourceToDelete(resource schema.GroupVersionResource, metadata
...
+	retriableAPIErrors := func(err error) bool {
+		return apierrors.IsServiceUnavailable(err) || // in case API server is rolling out
+			apierrors.IsServerTimeout(err) ||
+			apierrors.IsUnexpectedServerError(err) ||
+			netutil.IsConnectionRefused(err) ||
+			netutil.IsConnectionReset(err) ||
+			netutil.IsNoRoutesError(err) ||
+			netutil.IsProbableEOF(err)
+	}
+	if err := kretry.OnError(kretry.DefaultRetry, retriableAPIErrors, func() error {
```
Comment (sttts) [inline test/extended/util/client.go:731]: isn't there a retry func with context already?

File: test/extended/util/client.go (hunk 1 above)
Comment (mfojtik) [reply test/extended/util/client.go:739]: stacktrace, original code did this

File: test/extended/util/client.go (hunk 2 above)
Comment (mfojtik) [reply test/extended/util/client.go:731]: in our library-go, wanted to avoid unnecessary import

File: test/extended/util/client.go (hunk 1 above)
Comment (sttts) [reply test/extended/util/client.go:739]: k

File: test/extended/util/client.go (hunk 2 above)
Comment (sttts) [reply test/extended/util/client.go:731]: don't we have library-go here?

**Comment by deads2k** [comment]
why do we think we want to tolerate a 503 instead of trying to track down what is causing it to be unavailable?

**Comment by deads2k** [comment]
/hold

This is working around an azure specific problem.  We should not do this.

**Pull request closed without merging**